src/Microsoft.AspNetCore.DataProtection/ActivatorExtensions.cs

```diff
@@ -37,55 +37,8 @@ public static T CreateInstance<T>(this IActivator activator, string implementati
         public static IActivator GetActivator(this IServiceProvider serviceProvider)
         {
             return (serviceProvider != null)
-                ? (serviceProvider.GetService<IActivator>() ?? new SimpleActivator(serviceProvider))
-                : SimpleActivator.DefaultWithoutServices;
-        }
-
-        /// <summary>
-        /// A simplified default implementation of <see cref="IActivator"/> that understands
-        /// how to call ctors which take <see cref="IServiceProvider"/>.
-        /// </summary>
-        private sealed class SimpleActivator : IActivator
-        {
-            /// <summary>
-            /// A default <see cref="SimpleActivator"/> whose wrapped <see cref="IServiceProvider"/> is null.
-            /// </summary>
-            internal static readonly SimpleActivator DefaultWithoutServices = new SimpleActivator(null);
-
-            private readonly IServiceProvider _services;
-
-            public SimpleActivator(IServiceProvider services)
-            {
-                _services = services;
-            }
-
-            public object CreateInstance(Type expectedBaseType, string implementationTypeName)
-            {
-                // Would the assignment even work?
-                var implementationType = Type.GetType(implementationTypeName, throwOnError: true);
-                expectedBaseType.AssertIsAssignableFrom(implementationType);
-
-                // If no IServiceProvider was specified, prefer .ctor() [if it exists]
-                if (_services == null)
-                {
-                    var ctorParameterless = implementationType.GetConstructor(Type.EmptyTypes);
-                    if (ctorParameterless != null)
-                    {
-                        return Activator.CreateInstance(implementationType);
-                    }
-                }
-
-                // If an IServiceProvider was specified or if .ctor() doesn't exist, prefer .ctor(IServiceProvider) [if it exists]
-                var ctorWhichTakesServiceProvider = implementationType.GetConstructor(new Type[] { typeof(IServiceProvider) });
-                if (ctorWhichTakesServiceProvider != null)
-                {
-                    return ctorWhichTakesServiceProvider.Invoke(new[] { _services });
-                }
-
-                // Finally, prefer .ctor() as an ultimate fallback.
-                // This will throw if the ctor cannot be called.
-                return Activator.CreateInstance(implementationType);
-            }
+                ? (serviceProvider.GetService<IActivator>() ?? new SimpleActivator(serviceProvider.GetService<ILoggerFactory>()))
```

Comment:
In what case would we have no `IActivator` registered but `ILoggerFactory` registered? Is the activator a replaceable abstraction? (ask yourself if that makes sense).